lib/tasks/import.rake

```diff
@@ -0,0 +1,57 @@
+namespace :reports do
+  task :import => :environment do
+    CSV.foreach("logs.csv", headers: true) do |l|
+      log = Log.new
+      log.id = l['id']
+      log.source_id = l['source_id']
+      log.message_id = l['message_id']
+      log.report_id = l['report_id']
+      log.level_id = l['level_id']
+      User.without_auditing do
+        User.as_anonymous_admin do
+          log.save!
+        end
+      end
```

Comment:
This is a bit slow, try to wrap it into a single transaction:

    Log.transaction do ... end

Reply:
You need to wrap the whole loop in a transaction, that should speed things up.

The same for sources, messages, reports.